src/hevm/src/EVM/UnitTest.hs

```diff
@@ -577,8 +578,10 @@ symFailure UnitTestOptions {..} testName failures' = mconcat
   , intercalate "\n" $ indentLines 2 . mkMsg <$> failures'
   ]
   where
-    showRes vm = let Just res = view result vm
-                 in prettyvmresult res
+    showRes vm = let Just res = view result vm in
+                 case res of
+                   VMFailure _ -> prettyvmresult res
+                   VMSuccess _ -> "DSTest Assertion Violation"
```

Comment:
Unless I'm missing something, it should be safe to assume that all vm's at this point that have not failed are the result of a ds test assertion violation?

Reply:
Hmm, hard to see the context with this diff, but what about proveFail cases?

Reply:
v good point.

Reply:
Think I have this handled now.